usi/src/main/java/daq/usi/UsiImpl.java

```diff
@@ -19,7 +21,11 @@ public UsiImpl() {
     switchControllers = new HashMap<>();
   }
 
-  private SwitchController getSwitchController(SwitchInfo switchInfo) {
+  private ISwitchController getSwitchController(SwitchInput switchInput) {
+    if (!switchInput.hasSwitchInfo()) {
```

Comment:
Would be better to have an explicit value for the OVS variant. Keeping the door open to a) having no switch defined (no switch info) and then also having explicit switch info parameters for an OVS variant (e.g. how do we test the use of a password?).  E.g., just move this down to a case clause in the switch below. See config/system/ext.conf -- it uses FAUX_SWITCH for this -- so you could pick something similar. (I would avoid just "OVS" because it's too short, but OVS_SWITCH or something (even through it's redundant.)

This also then is creating a new OpenVSwitch object for each new request?  That doesn't seem right either.

Reply:
The 2 cases (with a physical switch and ovs) aren't at all similar which was why I had to make so many changes for ovs case to work. Only is faux interface is needed to be passed for the port connect and disconnect. As faux interface is a string(and not a device port), it doesn't match the existing signatures for all of the service functions. I decided to just pass the faux interface separately from the existing switchInfo and pass that information during object instantiation(the class is completely static so it hurt performance). The faux interface should be coming from DAQ; usi shouldn't take up the responsibility of parsing the config file.